test/unit/uk/gov/hmrc/apinotificationpull/controllers/NotificationsControllerSpec.scala

```diff
@@ -17,44 +17,80 @@
+import uk.gov.hmrc.apinotificationpull.model.Notifications
+import uk.gov.hmrc.apinotificationpull.services.ApiNotificationQueueService
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 
-class NotificationsControllerSpec extends UnitSpec with WithFakeApplication {
+import scala.concurrent.Future
+
+class NotificationsControllerSpec extends UnitSpec with WithFakeApplication with MockitoSugar {
+
+  private val notificationId1 = UUID.randomUUID()
+  private val notificationId2 = UUID.randomUUID()
+  private val notifications = Notifications(List(s"/notification/$notificationId1", s"/notification/$notificationId2"))
 
-  private val headerValidator = new SuccessfulHeaderValidatorFake
-  private val controller = new NotificationsController(headerValidator)
-  private val notificationId = UUID.randomUUID()
   private val xClientIdHeader = "X-Client-ID"
+  private val clientId = "client_id"
+
+  private val validHeaders = Seq(ACCEPT -> "application/vnd.hmrc.1.0+xml", xClientIdHeader -> clientId)
 
-  "delete notification by id" when {
-    val validRequest = FakeRequest("DELETE", s"/$notificationId").
-      withHeaders(ACCEPT -> "application/vnd.hmrc.1.0+xml", xClientIdHeader -> "client-id")
+  trait Setup {
+    implicit val materializer = fakeApplication.materializer
 
-    "notification does not exist" should {
-      "return 404 NOT_FOUND response" in {
-        val result = controller.delete(notificationId.toString).apply(validRequest)
+    val headerValidator = new SuccessfulHeaderValidatorFake
+
+    val mockApiNotificationQueueService = mock[ApiNotificationQueueService]
+    val controller = new NotificationsController(mockApiNotificationQueueService, headerValidator)
+  }
+
+  "delete notification" should {
+
+    val validRequest = FakeRequest("DELETE", s"/$notificationId1").withHeaders(validHeaders: _*)
+
+    "return 404 NOT_FOUND response when the notification does not exist" in new Setup {
+        val result = await(controller.delete(notificationId1.toString).apply(validRequest))
 
         status(result) shouldBe NOT_FOUND
+        bodyOf(result) shouldBe ""
       }
     }
-  }
 
-    "get all notification by client id" when {
+  "get all notifications" should {
 
-    // TODO: fix it after implementation is done
-    "implementation is not done yet" should {
-      "throw an exception because not implemented yet" in {
+    val validRequest = FakeRequest("GET", "/").withHeaders(validHeaders: _*)
 
-        val result = controller.getAll().apply(FakeRequest())
-        intercept[Exception] {
-          await(result)
-        }
-      }
+    "return all notifications" in new Setup {
+      when(mockApiNotificationQueueService.getNotifications()(any(classOf[HeaderCarrier])))
+        .thenReturn(Future.successful(notifications))
+
+      val result = await(controller.getAll().apply(validRequest))
+
+      status(result) shouldBe OK
+
```

Comment:
Think about converting `bodyOf(result)` to Scala XML type and comparing that with `expectedXml` created with Scala Xml DSL